source/frontend/src/features/documents/ComposedDocument.ts

```diff
@@ -23,16 +23,16 @@ export interface ComposedDocument {
 
 export class DocumentRow {
   id?: number;
-  mayanDocumentId?: number;
-  documentType?: Api_DocumentType;
-  statusTypeCode?: Api_TypeCode<string>;
-  fileName?: string;
-  isFileAvailable?: boolean;
+  mayanDocumentId: number | undefined;
```

Comment:
This article does a decent job of describing the differences, but essentially having an interface with all optional fields is the same as {}. Optional fields are not validated by ts.

this article discusses the differences
https://kate-dev.medium.com/when-to-use-typescript-optional-property-how-is-it-different-from-declaring-property-as-undefined-2319a0ee1f07

Essentially having an interface with all optional fields isn't much better than any. We probably need some tech debt to go back and fix this, and maybe a discussion on Friday about how/when we want ? vs | undefined